### PR TITLE
122 secret load test properties

### PR DESCRIPTION
**Pull request by DerNamenlose**
This PR adds support for secret properties to XLT. Secret Properties are defined by prefixing their name with `secret.`. They are available under their full name and without the `secret.` prefix. Their values are not written to the result set and are not rendered to any report.

### CODE REVIEW

File: src/main/java/com/xceptance/xlt/agentcontroller/AgentControllerImpl.java [hunk 1]
```diff
@@ -1063,6 +1070,76 @@ public String archiveTestConfig()
         return tempConfigArchiveFile != null ? tempConfigArchiveFile.getName() : null;
     }
 
+    /**
+     * Add a file of config directories with masked secrets to the output
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/agentcontroller/AgentControllerImpl.java:1074]:
```suggestion
     * Adds the properties files contained contained in the given config directory to the given ZIP output stream whereas secret properties will be written out masked.
```

File: src/main/java/com/xceptance/xlt/agentcontroller/AgentControllerImpl.java [hunk 2]
```diff
@@ -1063,6 +1070,76 @@ public String archiveTestConfig()
...
+        while (files.hasNext())
+        {
+            final File fileToMask = files.next();
+            if (fileToMask.getAbsolutePath().endsWith(XltConstants.PROPERTY_FILE_EXTENSION))
+            {
+                maskFile(fileToMask, fileToMask);
+            }
+        }
+        ZipUtils.zipDirectory(out, tempDir, filter, configPath);
+        FileUtils.deleteDirectory(tempDir);
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/agentcontroller/AgentControllerImpl.java:1097]:
```suggestion
        try 
        {
            FileUtils.copyDirectory(configDirectory, tempDir, filter);
            final Iterator<File> files = FileUtils.iterateFiles(tempDir, null, true);
            while (files.hasNext())
            {
                final File fileToMask = files.next();
                if (fileToMask.getAbsolutePath().endsWith(XltConstants.PROPERTY_FILE_EXTENSION))
                {
                    maskFile(fileToMask, fileToMask);
                }
            }
            ZipUtils.zipDirectory(out, tempDir, filter, configPath);
        }
        finally
        {
            FileUtils.deleteDirectory(tempDir);
        }
```

File: src/main/java/com/xceptance/xlt/agentcontroller/AgentControllerImpl.java [hunk 3]
```diff
@@ -1063,6 +1070,76 @@ public String archiveTestConfig()
...
+    private static void maskFile(File inputFile, File outputFile) throws ConfigurationException, IOException
+    {
+        final FileReader reader = new FileReader(inputFile);
+        PropertiesConfiguration config = new PropertiesConfiguration();
+        config.read(reader);
+        reader.close();
+        config = mask(config, inputFile.getName().equals(XltConstants.SECRET_PROPERTIES_FILENAME));
+        final StringWriter writer = new StringWriter();
+        config.write(writer);
+        FileUtils.writeStringToFile(outputFile, writer.toString(), Charset.defaultCharset());
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/agentcontroller/AgentControllerImpl.java:1119]:
```suggestion
        final PropertiesConfiguration inputConfig = new PropertiesConfiguration();
        try (final FileReader reader = new FileReader(inputFile))
        {
            inputConfig.read(reader);
        }   
        
        final PropertiesConfiguration outputConfig = mask(inputConfig, inputFile.getName().equals(XltConstants.SECRET_PROPERTIES_FILENAME));
        try (final StringWriter writer = new StringWriter())
        {
            outputConfig.write(writer);
        }
        // write out masked properties (ensure correct encoding)
        FileUtils.writeStringToFile(outputFile, writer.toString(), StandardCharsets.ISO_8859_1);
```

File: src/test/java/com/xceptance/xlt/report/providers/ConfigurationReportProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * Unit tests for the {@link ConfigurationReportProvider}
+ */
+public class ConfigurationReportProviderTest
+{
+    @Test
+    public void testSecretPropertiesAreMaskedInTheOutput() throws IOException
+    {
+        final ConfigurationReportProvider provider = new ConfigurationReportProvider();
+        ReportGeneratorConfiguration config = new ReportGeneratorConfiguration();
+        config.setReportDirectory(new File("samples/testsuite-posters"));
```
Comment (h-arlt) [inline src/test/java/com/xceptance/xlt/report/providers/ConfigurationReportProviderTest.java:37]: This test is likely to fail for standard XLT installations as it assumes local modifications of the shipped testsuite.

File: src/main/java/com/xceptance/xlt/report/providers/ConfigurationReportProvider.java [hunk 5]
```diff
@@ -42,7 +42,7 @@
 import com.xceptance.xlt.util.XltPropertiesImpl;
 
 /**
- * 
+ * Report provider generating a filtered list of the configuration used for the test run
```
Comment (h-arlt) [inline src/main/java/com/xceptance/xlt/report/providers/ConfigurationReportProvider.java:45]:
```suggestion
 * Report provider generating a report fragment about the configuration used for the test run
```

**Review by h-arlt — COMMENTED** [review]
Please see comments inline.

File: src/test/java/com/xceptance/xlt/report/providers/ConfigurationReportProviderTest.java (hunk 4 above)
Comment (DerNamenlose) [reply src/test/java/com/xceptance/xlt/report/providers/ConfigurationReportProviderTest.java]: Fixed by generating the appropriate directory layout.

**Review by h-arlt — APPROVED** [review]
Looks good to me.

File: src/main/java/com/xceptance/xlt/util/XltPropertiesImpl.java [hunk 6]
```diff
@@ -75,8 +77,13 @@
     private String version;
 
     /**
-     * Contains the absolute paths to the resolved property files. This means the property files which are there by
-     * default and the property files transitively included by &quot;includes&quot; in these property files.
+     * The prefix used for secret properties
+     */
+    private final String SECRET_PREFIX = "secret.";
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/util/XltPropertiesImpl.java:82]: This constant is already defined in XltConstants. Can we use that one?

**Pull request merged**